genai-perf/genai_perf/inputs/converters/tensorrtllm_engine_converter.py

```diff
@@ -82,4 +82,35 @@ def _add_request_params(self, payload: Dict, config: InputsConfig) -> None:
                 payload["min_length"] = [num_tokens]
 
         for key, value in config.extra_inputs.items():
-            payload[key] = [value]
+            if key == "set_end_id":
+                payload["end_id"] = [config.tokenizer._tokenizer.eos_token_id]
+            elif key == "apply_chat_template":
+                pass
+            else:
+                payload[key] = [value]
+
+    def _encode_tokens(self, prompt: str, config: InputsConfig) -> List[int]:
+        if config.extra_inputs.__contains__("apply_chat_template"):
+            token_ids = self._encode_with_chat_template(prompt, config)
+        else:
+            token_ids = config.tokenizer.encode(prompt, add_special_tokens=False)
+        return token_ids
+
+    def _encode_with_chat_template(
+        self, prompt: str, config: InputsConfig
+    ) -> List[int]:
+        """
+        Apply the default TRT-LLM engine chat template to the prompt
+        """
+        try:
+            import jinja2
+        except ImportError:
+            raise ImportError(
+                "Jinja2 is required for using TRT-LLM with chat template processing. Install it using: pip install jinja2."
+            )
+        default_template = [{"role": "user", "content": prompt}]
```

Comment:
Shouldn't we move this to a more global location for easier overriding of the default template if we want to? This being so local here feels a bit wrong.

Reply:
Yeah, I was unsure about this. It's specific to this converter. I can move it to the top of the class if that's helpful. Let me do that now.

Reply:
This is the only place this can go, because the prompt is a local value. Do you prefer a different approach?

Reply:
Maybe potentially a method to construct it so it's more clear? I'm not too hung out about this one.

Reply:
How's this? https://github.com/triton-inference-server/perf_analyzer/pull/253/commits/d693c583d7f93c45e283ca7b230406e2fb3c03e7